enhancements/microshift/feature-csi-snapshots.md

```diff
@@ -0,0 +1,566 @@
+enable global configuration options for their storage back end.
+
+```yaml
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshotClass
+metadata:
+  name: csi-hostpath-snap
+driver: hostpath.csi.k8s.io [1] 
+deletionPolicy: Delete
+```
+
+1. The name of the CSI driver that is used to create snapshots of this **VolumeSnapshotClass** object. The name 
+must be the same as the Provisioner field of the storage class that is responsible for the PVC that is being 
+snapshotted.
+
+#### VolumeSnapshot
+
+Similar to the PersistentVolumeClaim object, the VolumeSnapshot CRD defines a developer request for a snapshot. The 
+CSI snapshot controller handles the binding of a VolumeSnapshot CRD with an appropriate VolumeSnapshotContent CRD. 
+The binding is a one-to-one mapping.
+
+The VolumeSnapshot CRD is namespaced. A developer uses the CRD as a distinct request for a snapshot.
+
+```yaml
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshot
+metadata:
+  name: mysnap
+spec:
+  volumeSnapshotClassName: csi-hostpath-snap [1] 
+  source:
+    persistentVolumeClaimName: myclaim [2]
+```
+
+1. The request for a particular class by the volume snapshot. If the **volumeSnapshotClassName** setting is absent 
+and there is a default volume snapshot class, a snapshot is created with the default volume snapshot class name. 
+But if the field is absent and no default volume snapshot class exists, then no snapshot is created.
+2. The name of the **PersistentVolumeClaim** object bound to a persistent volume. This defines what you want to 
+create a snapshot of. Required for dynamically provisioning a snapshot.
+
+#### VolumeSnapshotContent
+
+A snapshot taken of a volume in the cluster that has been provisioned by a cluster administrator.
+
+Similar to the PersistentVolume object, the VolumeSnapshotContent CRD is a cluster resource that
+points to a real snapshot in the storage back end.
+
+For manually pre-provisioned snapshots, a cluster administrator creates a number of
+VolumeSnapshotContent CRDs. These carry the details of the real volume snapshot in the storage system.
+
+The VolumeSnapshotContent CRD is not namespaced and is for use by a cluster administrator.
+
+```yaml
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshotContent
+metadata:
+  name: new-snapshot-content-test
+  annotations:
+    - snapshot.storage.kubernetes.io/allow-volume-mode-change: "true"
+spec:
+  deletionPolicy: Delete
+  driver: hostpath.csi.k8s.io
+  source:
+    snapshotHandle: 7bdd0de3-aaeb-11e8-9aae-0242ac110002
+  sourceVolumeMode: Filesystem
+  volumeSnapshotRef:
+    name: new-snapshot-test
+    namespace: default
+```
+
+### Risks and Mitigations
+
+- Increased overhead.  The CSI snapshot controller and sidecar increase the on-disk footprint by a
+total of 360Mb.  At idle, the components use a negligible amount of CPU (>1% CPU on a 2 core machine)
+and roughly 18Mb.
```

Comment:
How much is 360MB from the current MicroShift container footprint?

Reply:
A quick `df` before and after installing MicroShift on a fresh system shows an image footprint of just over 2Gb. So this is an increase of about an 18% on-disk footprint.